### PR TITLE
New slot level events

### PR DESCRIPTION
**Pull request by carlesandres**
This PR introduces the following changes:

1. Several custom events fired by `o-ads` have been given more meaningful names.
2. `o-ads` is now recording a [performance mark](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) for every event that it dispatches. It's worth mentioning that:
   1. The performance mark will have the same name as the event that originated it.
   1. If the event has a payload the `pos`, `size` and `name` will be appended to the event name to construct the performance mark name so that those details can be retrieved later and used for monitoring purposes. An example of such type of performance mark name is `oAds.slotExpand__mid__halfpage-desktop__300,600`.
3. `o-Ads` exposes a new method called `setupMetrics` that accepts the following parameters:
   1. An array of objects containing the configuration of the different metrics that need to be tracked. 
   1. A callback function to be called whenever any of the `trigger` events of any of those configuration objects is dispatched.

This will enable the following:

1. Monitoring the slot rendering phase.
2. Remove monitoring code from the consuming apps (`n-ui`, `ft-app`, etc.)

### CODE REVIEW

File: src/js/ad-servers/gpt.js [hunk 1]
```diff
@@ -340,7 +349,7 @@ const slotMethods = {
 	*/
 	display: function() {
 	  window.googletag.cmd.push(() => {
-	    utils.broadcast('gptDisplay');
+			this.fire('slotGoRender');
```
Comment (SamiTriki) [inline src/js/ad-servers/gpt.js]: Indentation

File: src/js/ad-servers/gpt.js [hunk 2]
```diff
@@ -238,6 +239,13 @@ function onRenderEnded(event) {
 	const iframeId = `google_ads_iframe_${gptSlotId.getId()}`;
 	data.type = domId.pop();
 	data.name = domId.join('-');
+	data.size = event.size && event.size.length ? event.size.toString() : '';
+
+	const slotTargeting = event.slot.getTargetingMap && event.slot.getTargetingMap();
+	if (slotTargeting && slotTargeting.pos) {
+		data.pos = slotTargeting.pos.length ? slotTargeting.pos.join('') : '';
```
Comment (carlesandres) [inline src/js/ad-servers/gpt.js]:
```suggestion
		data.pos = slotTargeting.pos.length ? slotTargeting.pos.join(',') : '';
```

File: src/js/ad-servers/gpt.js [hunk 3]
```diff
@@ -238,6 +239,13 @@ function onRenderEnded(event) {
 	const iframeId = `google_ads_iframe_${gptSlotId.getId()}`;
 	data.type = domId.pop();
 	data.name = domId.join('-');
+	data.size = event.size && event.size.length ? event.size.toString() : '';
```
Comment (carlesandres) [inline src/js/ad-servers/gpt.js]:
```suggestion
	data.size = event.size && event.size.length ? event.size.join() : '';
```

File: package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "test-cy:open": "cypress open",
     "test-nw": "npm run test-nw:basic && npm run test-nw:extended",
     "test-nw:local": "nightwatch -c ./test/nightwatch/config/nightwatch.conf.local.js",
-    "test-nw:basic": "npm run nightwatch-bs -- --group basic --env chrome,edge,galaxy_s8",
+    "test-nw:basic": "npm run nightwatch-bs -- --group basic --env chrome,edge",
```
Comment (alevito) [inline package.json]: Maybe worth adding a github issue or a TODO comment to remember that galaxy_s8 has been temporarly removed and we need to look into it in the future?

Comment (carlesandres) [reply package.json]: I have already created a ticketon the BAU board: https://trello.com/c/kyIWyQ15/393-fix-problem-with-samsung-galaxy-testing-in-o-ads

File: src/js/ad-servers/gpt.js (hunk 1 above)
Comment (carlesandres) [reply src/js/ad-servers/gpt.js]:
```suggestion
		this.fire('slotGoRender');
```

File: src/js/ad-servers/gpt.js (hunk 3 above)
Comment (alevito) [reply src/js/ad-servers/gpt.js]: Calling join() on an empty - zero lenght - array returns an empty string, so we can aboid to check for the array length
```suggestion
	data.size = event.size && event.size.join()';
```

File: src/js/ad-servers/gpt.js (hunk 2 above)
Comment (alevito) [inline src/js/ad-servers/gpt.js]:
```suggestion
		data.pos = slotTargeting.pos.join();
```

Comment (carlesandres) [reply src/js/ad-servers/gpt.js]: @alevito slotTargeting.pos might be null or undefined in some case. Checking for `.length` is defensive code to "ensure" it's an Array

**Pull request merged**